### ISSUE TITLE
[Error] Failed to activate, Error while adding the certificates to AMT

### ISSUE THREAD

**Issue by michaelmumenthaler** [open-amt-cloud-toolkit/open-amt-cloud-toolkit#139]
**Describe the bug** 🪲
Hi everyone,

We ran into an issue when trying to join a new device to AMT with rpc.exe. We have 2 new devices and we get the same error on both devices. The device in question is an HP Z2 Mini G9. The older models work as intended, so I'm unsure if it's even AMT related or if something with the device is wrong. Nevertheless I hope you might be able to point me to the right direction.

We are trying to activate the devices with admin mode (cert and config all works on the other devices). When we run the RPC command we get the following error on the device: 

time="2022-08-16T09:06:38+02:00" level=error msg="{\"Status\":\"Device 1cf33dee-9ccf-49d9-81fe-112f10acb322 activation failed. Error while adding the certificates to AMT.\"}"

RPS logs show the same error, not more information is provided there. 

I ran the rpc command with verbose output but I don't see anything in there that points me to the problem. 
On the troubleshooting guide in your docs it is said to disconnect the device from power and LAN, so I did this over the weekend to no avail. 

We have 2 OACT instances (Dev & Prod) and the error is the same on both. 

The devices that work have AMT version 14.1.65, where the new devices have 16.0.15. I'm not sure if this might be the cause of the issue.

Thank you in advance for any information on this issue.

**To Reproduce** 🪜
1. run rpc.exe with the relevant parameters
2. see the error

**Expected behavior**
The device should be able to join AMT. 

**Screenshots** 🖼️
None

**AMT Device (please complete the following information):** 🖥️
 - OS: Windows 10 21H2 Build 19044.1889
 - AMT Version: 16.0.15
 - AMT Configuration Mode: Admin Control Mode
 - Network Configuration: Static

**Service Deployment (please complete the following information):** ⛈️
 - Deployment Type: Docker
 - Node Version: Ubuntu LTS 22.04
 - Component & Version: 2.4

**Additional context**

**Comment by michaelmumenthaler** [comment]
Sidenote: Activating AMT in Client Control Mode works.

**Comment by graikhel-intel** [comment]
Hi @michaelmumenthaler,

Could you please provide the logs from RPS server so I can take a look at the issue? change `RPS_LOG_LEVEL` in [.env](https://github.com/open-amt-cloud-toolkit/open-amt-cloud-toolkit/blob/main/.env.template#L8) file to `silly` and re-run the RPS to capture logs

Also, could you run `rpc.exe amtinfo` command on the AMT 16.0.15 device and attach the output to this issue?

Just making sure that AMT device is on DHCP and not static network? Currently, ACM activation is only supported when AMT platform is on DHCP.

I speculate it has to do something with the provisioning certificate or the network configuration but can narrow it down better once I see RPS logs.

**Comment by michaelmumenthaler** [comment]
Hi @graikhel-intel 

Thanks for the reply.

I attached the logs (AMT Info & RPS Logs in silly mode). I redacted some sensitive information (cert & passwords).
[amtinfo.txt](https://github.com/open-amt-cloud-toolkit/open-amt-cloud-toolkit/files/9414169/amtinfo.txt)
[RPS-LOG.txt](https://github.com/open-amt-cloud-toolkit/open-amt-cloud-toolkit/files/9414172/RPS-LOG.txt)

The network config in the AMT profile is set to DHCP and the client is also set to obtain an IP from DHCP. 

If i run the exact same configuration on an older device with AMT version 14.1.65 it works without any issues.

**Comment by graikhel-intel** [comment]
Thanks for the logs @michaelmumenthaler 

Support for SHA1 Root certificates has been removed starting from Intel ME 15.0 firmware for H platform, and Intel ME 16.0 firmware for all platforms https://software.intel.com/sites/manageability/AMT_Implementation_and_Reference_Guide/default.htm?turl=WordDocuments%2Frootcertificatehashes.htm

AMT 16 device is throwing [cert_verify_failed](https://software.intel.com/sites/manageability/AMT_Implementation_and_Reference_Guide/default.htm?turl=HTMLDocuments%2FWS-Management_Class_Reference%2FIPS_HostBasedSetupService.htm%23AddNextCertInChain) error (or ReturnValue 4) because the root certificate is generated using SHA1 signature hash algorithm. That explains why it is working on AMT 14.x and failing on AMT 16.x.  I've tested the same scenario using SHA1 root certificate & AMT 16 device, and was able to see the same error.

GoDaddy (cert issuing team) should be able to help you with creating full SHA2 certificate chain or I can email you privately and add a colleague who can help with fixing certificate issue. Let me know if that works.
